libkernelflinger/log.c

```diff
@@ -163,11 +170,51 @@ void vlog(const CHAR16 *fmt, va_list args)
 void log(const CHAR16 *fmt, ...)
```

Comment:
All of this implementation is typical of the very poor quality of the patch made on Kernelflinger the last three years !

You should create two functions:
1. `log_serial()`
2. `log_file()`

Then you make the `log()` function or even better the `log` symbol point to the implementation you want.

Reply:
I only create one functions:
void log_file (const CHAR16 *fmt, va_list args);

I don't think it's necessary to create another function. It's only two lines and doesn't need to change the original idea.